### ISSUE TITLE
Text-shadow removes rgb values when element selected

### ISSUE THREAD

**Issue by benvmatheson** [artf/grapesjs#2366]
Example:
https://jsfiddle.net/benvmatheson/g1ft06qh/8/

Import a style with a text property.
`text-shadow:rgb(51, 166, 58) 4px 4px 4px;`

This works correctly, and the item is displayed correctly. However, once the item is selected, the rgb value is dropped and replaced with 0.
`text-shadow:0 4px 4px 4px;`

**Comment by artf** [comment]
Yeah, I'm already aware of this issue. 
Unfortunately, the Style Manager parser is quite dumb and what it does is simply splitting the value by a separator, in this case `' '` (space), and distributing splits to relative properties by their order in the definition. So, technically, this change would make it work `4px 4px 4px rgb(51, 166, 58)`, but... unfortunately it doesn't and this is due to [CSSOM inconsistency](https://grapesjs.com/docs/guides/Custom-CSS-parser.html#cssom-results-are-inconsistent). 
Probably the "best solution", in this case, would be to change the text-shadow property definition but, to be honest, I'd prefer to find a more solid solution. A bit smarter parser might be a good start... so, I'm open to any suggestion/help to this issue

**Comment by benvmatheson** [comment]
This seems like it could turn into a big project due to cross browser dependencies. Would it be worth bringing in a 3rd party module instead of trying to solve this inside grapes?

**Comment by benvmatheson** [comment]
I'm also somewhat confused about where the parser is taking effect. The style shows correctly when first loaded. It's not until you click the element that the style disappears. Would this be the same issue?

**Comment by benvmatheson** [comment]
Also, it appears to fail whether the `rgb()` comes at the first or last. And it appears to succeed with hex colors. This may be a separate issue?

**Comment by benvmatheson** [comment]
@artf It doesn't look like anything within ParserCss.js or BrowserParserCss.js is being run when the element is clicked and the style is removed. Do you know where else I would look to see what could be editing rules when the element is clicked?

**Comment by artf** [comment]
When you change the component selection, all properties in style manager are triggered from [this method](https://github.com/artf/grapesjs/blob/4bc79bd304072b458f63482d3a4f5049c6491467/src/style_manager/view/PropertyView.js#L209)... in case of text-shadow, being of `stack` type, the method is [this one](https://github.com/artf/grapesjs/blob/4bc79bd304072b458f63482d3a4f5049c6491467/src/style_manager/view/PropertyStackView.js#L38)

**Comment by benvmatheson** [comment]
It looks like `targetUpdated` is run twice when the element is clicked. Once with the correct value `rgb(89, 89, 89) 5px 6px 7px` and a second time with `rgb` cleared `0 5px 6px 7px. This would explain why it's confused whether the first or last number should be interpreted as the color.

Any ideas where this second run would come from?

**Comment by artf** [comment]
I'm closing this as now it should be possible to extend the [built-in properties](https://grapesjs.com/docs/modules/Style-manager.html#programmatic-usage) and by extending the Stack type property is also possible to use `fromStyle` in order to implement custom logic for parsing (example here: https://grapesjs.com/docs/api/property_composite.html#propertycomposite)
